core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1973,6 +1971,7 @@ public static ExpressionAnalysis analyzeExpression(
         analyzer.analyze(expression, scope);
 
         updateAnalysis(analysis, analyzer, session, accessControl);
+        analysis.addReferencedFields(expression, analyzer.getReferencedFields());
```

Comment:
Probably `referencedFields` could be now simplified to `Set<Field>`. I don't think the mapping by source node is used.